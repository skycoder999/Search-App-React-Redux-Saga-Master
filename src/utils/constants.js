export const API_SEARCH_PATH = "https://api.themoviedb.org/3/search/movie?api_key=f9dba24a3b8ed9425600eb5d5fbd9a93";
export const API_POSTER_PATH = "https://image.tmdb.org/t/p/w92";
export const API_POSTER_PATH_BIG = "https://image.tmdb.org/t/p/w500";
export const API_THROTTLE_DELAY = 300;
export const API_RESULTS_LIMIT = 10;
export const TEXT_SEARCH_PLACEHOLDER = "Type here to search movies...";
export const TEXT_LOADING = "Loading results...";
export const TEXT_NO_RESULTS = "Oops! No results found.";

