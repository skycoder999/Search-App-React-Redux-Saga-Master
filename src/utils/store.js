import { createStore, applyMiddleware } from "redux";
import { logger } from "redux-logger";
import createSagaMiddleware from "redux-saga";
import rootReducer from "../reducers/root.reducer";
import rootSaga from "../sagas/root.saga";

const configureStore = (initialState = {}) => {
  const DEBUG = false;
  const sagaMiddleware = createSagaMiddleware();
  const middlewares = [sagaMiddleware, DEBUG && logger];
  const store = createStore(rootReducer, initialState, applyMiddleware(...middlewares.filter(Boolean)));
  sagaMiddleware.run(rootSaga);
  return store;
};

export default configureStore;
