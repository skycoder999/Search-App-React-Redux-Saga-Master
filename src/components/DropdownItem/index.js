import React from "react";
import PropTypes from "prop-types";
import { API_POSTER_PATH } from "../../utils/constants";

const DropdownItem = ({ item, onSelect }) => {
  return (
    <div className="suggest-item" title={item.overview} onClick={() => onSelect(item)}>
      {item.poster_path ? <img className="suggest-item__poster" src={`${API_POSTER_PATH}${item.poster_path}`} alt="" /> : <span className="suggest-item__poster" />}
      <div className="suggest-item__info">
        {!!item.vote_average ? <b>â˜… {Number(item.vote_average).toFixed(1)}</b> : null}
        <h3 className="suggest-item__title">
          {item.original_title}
          <span>({item.release_date.split("-")[0]})</span>
        </h3>
        <p>{item.overview}</p>
      </div>
    </div>
  );
};

DropdownItem.propTypes = {
  item: PropTypes.shape({
    overview: PropTypes.string.isRequired,
    poster_path: PropTypes.string.isRequired,
    vote_average: PropTypes.number.isRequired,
    release_date: PropTypes.string.isRequired
  }),
  onSelect: PropTypes.func
};

export default DropdownItem;
