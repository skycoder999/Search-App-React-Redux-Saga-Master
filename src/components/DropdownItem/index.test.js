import DropdownItem from "./index";
import { API_POSTER_PATH } from "../../utils/constants";

let wrapper, consoleError;
let item = {
  id: 1,
  overview: "Sample Overview",
  original_title: "Sample Title",
  poster_path: "some_poster.jpg",
  vote_average: 4,
  release_date: "2019-02-01"
};

beforeEach(() => {
  wrapper = shallow(<DropdownItem item={item} />);
  consoleError = console.error;
  console.error = jest.spyOn(console, "error");
});

afterEach(() => {
  console.error = consoleError;
});

describe("SuggestionItem Component", () => {
  it("renders without crashing", () => {
    expect(wrapper.exists()).toBe(true);
  });

  it("matches the snapshot", () => {
    expect(wrapper).toMatchSnapshot();
  });
});
