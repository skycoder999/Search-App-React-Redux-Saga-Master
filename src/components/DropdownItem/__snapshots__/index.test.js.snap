// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`SuggestionItem Component matches the snapshot 1`] = `ShallowWrapper {}`;
